serverless/networking/serverless-ossm.adoc

```diff
@@ -0,0 +1,126 @@
+include::modules/serverless-document-attributes.adoc[]
+include::modules/ossm-document-attributes.adoc[]
+[id="serverless-ossm"]
+= Using {ProductShortName} with {ServerlessProductName}
+:context: serverless-ossm
+include::modules/common-attributes.adoc[]
+
+toc::[]
+
+// what benefits does this integration provide?
+
+.Prerequisites
+. Install xref:../../serverless/installing_serverless/installing-openshift-serverless.adoc#installing-openshift-serverless[{ServerlessProductName}].
+. Install xref:../../service_mesh/service_mesh_install/preparing-ossm-installation.adoc#preparing-ossm-installation[{ProductName}].
+
+.Procedure
+. Add the `default` namespace to the xref:../../service_mesh/service_mesh_install/installing-ossm.adoc#ossm-member-roll-create_installing-ossm[ServiceMeshMemberRoll] as a member:
++
+
+[source,yml]
+----
+apiVersion: maistra.io/v1
+kind: ServiceMeshMemberRoll
+metadata:
+  name: default
+  namespace: istio-system
+spec:
+  members:
+    - default
+----
++
+[IMPORTANT]
+====
+Adding sidecar injection to Pods in system namespaces such as `knative-serving` and `knative-serving-ingress` is not supported.
+====
+
+. Create a network policy that permits traffic flow from Knative system Pods to your Knative applications:
+.. Enter the command:
++
+
+[source,terminal]
+----
+$  oc label namespace knative-serving serving.knative.openshift.io/system-namespace=true --overwrite
+----
+.. Enter the command:
++
+
+[source,terminal]
+----
+$ oc label namespace knative-serving-ingress serving.knative.openshift.io/system-namespace=true --overwrite
+----
+
+.. Copy the following YAML into a `.yaml` file:
++
+
+[source.yml]
+----
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-from-serving-system-namespace
+  namespace: default
+spec:
+  ingress:
+  - from:
+    - namespaceSelector:
+        matchLabels:
+          serving.knative.openshift.io/system-namespace: "true"
+  podSelector: {}
+  policyTypes:
+  - Ingress
+----
+// double check this indentation; doesn't show up as YAML syntax highlighting
+
+.. Apply the `.yaml` file:
++
+[source,terminal]
+----
+$ oc apply -f <FILENAME>
+----
+
+. Enable sidecar injection to your Knative Service by adding the `sidecar.istio.io/inject="true"`` annotation to the Service resource:
++
+
+[source,yml]
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: hello
+spec:
+  template:
+    metadata:
+      name: hello
```

Comment:
It produces following the error:

```
$ oc apply -f a.yaml 
Error from server (BadRequest): error when creating "a.yaml": admission webhook "validation.webhook.serving.knative.dev" denied the request: validation failed: invalid value: "hello" must have prefix "hello-": spec.template.metadata.name
```

This `spec.template.metatadata.name` must be `hello-` like `hello-1`.
Or we can remove the line (`name: hello`) to create random name.

Reply:
Removed the line to keep it simple, but lmk if you think it's better to change the name.